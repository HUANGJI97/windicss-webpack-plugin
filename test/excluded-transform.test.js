"use strict";

const getModuleSource = require('./helpers/getModuleSource')
const vueWebpackCompiler = require ('./helpers/vueWebpackCompiler')

describe("Excluded transform", function() {

  it('should render vue', done => {
    const compiler = vueWebpackCompiler('excluded-transform', {
      scan: {
        extraTransformTargets: {
          detect: [
            i => {
              console.log('template --> ', i)
              return false
            }
          ],
          css: [
            (resource) => {
              return resource.indexOf('node_modules_demo/my-package/should-transform.css') >= 0
            }

          ]
        }
      }
    })
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])

      const toNotTransformCss = getModuleSource('should-not-transform', stats)
      expect(toNotTransformCss).toContain('theme(')
      expect(toNotTransformCss).toMatchSnapshot('should not transform')

      // check the windi file has generated the right classes
      const toTransformCss = getModuleSource('should-transform', stats)
      expect(toTransformCss).not.toContain('theme(')
      expect(toTransformCss).toMatchSnapshot('should transform')

      done(err)
    });
  });
});
