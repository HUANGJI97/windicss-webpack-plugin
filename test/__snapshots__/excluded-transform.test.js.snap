// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Excluded transform should render vue: should not transform 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\":root {\\\\n  --header-height: theme('spacing.14');\\\\n  --docs-scroll-margin-block: calc(var(--header-height) + 4rem);\\\\n  --blogpost-scroll-margin-block: calc(var(--header-height));\\\\n}\\\\n\\\\n@screen md {\\\\n  :root {\\\\n    --header-height: theme('spacing.18');\\\\n    --blogpost-scroll-margin-block: calc(var(--header-height) - 0.5rem);\\\\n  }\\\\n}\\\\n\\\\n@screen xl {\\\\n  :root {\\\\n    --docs-scroll-margin-block: calc(var(--header-height) + 1rem);\\\\n  }\\\\n}\\\\n\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;

exports[`Excluded transform should render vue: should transform 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\":root {\\\\n  --header-height: 3.5rem;\\\\n  --docs-scroll-margin-block: calc(var(--header-height) + 4rem);\\\\n  --blogpost-scroll-margin-block: calc(var(--header-height));\\\\n}\\\\n@media (min-width: 768px) {\\\\n  :root {\\\\n    \\\\n    --blogpost-scroll-margin-block: calc(var(--header-height) - 0.5rem);\\\\n  }\\\\n}\\\\n@media (min-width: 1280px) {\\\\n  :root {\\\\n    --docs-scroll-margin-block: calc(var(--header-height) + 1rem);\\\\n  }\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;
